lib/masamune/actions/transform.rb

```diff
@@ -77,13 +66,35 @@ def relabel_dimension(target)
     end
 
     def load_fact(source_files, source, target, date)
-      transform = Wrapper.load_fact(source_files, source, target, date)
-      postgres file: transform.to_file, debug: (source.debug || target.debug)
+      optional_apply_map(source_files, source, target) do |intermediate_files, intermediate|
+        transform = Wrapper.load_fact(intermediate_files, intermediate, target, date)
+        postgres file: transform.to_file, debug: (source.debug || target.debug || intermediate.debug)
+      end
     end
 
     def rollup_fact(source, target, date)
       transform = Wrapper.rollup_fact(source, target, date)
       postgres file: transform.to_file, debug: (source.debug || target.debug)
     end
+
+    private
+
+    def optional_apply_map(source_files, source, target, &block)
+      if source.respond_to?(:map) and map = source.map(to: target)
+        apply_map(map, source_files, source, target, &block)
+      else
+        yield source_files, source
+      end
+    end
+
+    def apply_map(map, source_files, source, target)
+      Tempfile.open('masamune') do |output|
+        FileUtils.chmod(FILE_MODE, output.path)
+        result = map.apply(source_files, output)
+        result.debug = map.debug
+        logger.debug(File.read(output)) if (source.debug || result.debug)
+        yield output, result
+      end
+    end
```

Comment:
good call - for some reason I always think `Tempfile`s are removed when they go out of scope